File: Platform/Database/Allors.Database.Protocol/Api/Pull/PullResponse.cs
// <copyright file="PullResponse.cs" company="Allors bvba">
// Copyright (c) Allors bvba. All rights reserved.
// Licensed under the LGPL license. See LICENSE file in the project root for full license information.
// </copyright>

namespace Allors.Protocol.Remote.Pull
{
    using System.Collections.Generic;

    public class PullResponse
    {
        public string[][] Objects { get; set; }

        public Dictionary<string, string> NamedObjects { get; set; }

        public Dictionary<string, string[]> NamedCollections { get; set; }

        public Dictionary<string, string> NamedValues { get; set; }
    }
}

File: Platform/Workspace/Typescript/framework/protocol/pull/PullResponse.ts
import { Response } from '../Response';

export interface PullResponse extends Response {
  namedCollections?: { [id: string]: string[]; };
  namedObjects?: { [id: string]: string; };
  namedValues?: { [id: string]: any; };

  accessControls?: string[][];
  objects?: string[][];
}
